src/index.js

```diff
@@ -158,35 +155,28 @@ export default function graphqlHTTP(options: Options): Middleware {
       query = params.query;
       variables = params.variables;
       operationName = params.operationName;
+      documentID = params.documentID;
 
       // If there is no query, but GraphiQL will be displayed, do not produce
       // a result, otherwise return a 400: Bad Request.
-      if (!query) {
+      if (!query && !documentID) {
         if (showGraphiQL) {
           return null;
         }
-        throw httpError(400, 'Must provide query string.');
+        throw httpError(400, 'Must provide query string or document ID.');
```

Comment:
Should probably only say document ID should be provided if `loadPersistedDocument` is available